Added config options for systemd-coredump functionality.

auntieNeo opened the pull request:
This adds simple support for configuring the management of core dumps through systemd to the NixOS config. I was debugging a systemd service with a segmentation fault when I noticed that /etc/systemd/coredump.conf isn't configurable through NixOS. This feature is helpful as systemd can place core dumps in a central location and rotate them as needed.

I left the default as systemd core dumps disabled, which is the current behavior. Limits still have to be set separately, which I believe is appropriate considering core dumps can be a security risk on sensitive systems.

I would have added the explicit --enable-coredump flag to the systemd configure flags, but decided against it when Nix started rebuilding Chromium etc. Someone might want to un-comment it at a later time.

auntieNeo committed:
Added config options for systemd-coredump functionality.

lethalman commented on pkgs/os-specific/linux/systemd/default.nix:
```diff
@@ -56,6 +56,9 @@ stdenv.mkDerivation rec {
       "--disable-split-usr"
+# FIXME: Can someone kindly uncomment this with the next systemd version bump?
+# I don't want to force anyone to rebuild systemd for a cosmetic change.
+#      "--enable-coredump"
```
Uncomment, we'll merge into staging.

bjornfor commented:
IMHO, since coredump support is on by default, we don't need "--enable-coredump". (I think it is impractical to be explicit about _all_ configure flags, so better take advantage of some defaults.)

I don't have strong feelings about that flag, but if you split this PR in two commits, one for the nixos options and one for the "--enable-coredump" flag, we can merge the nixos part quicker (no need to go via staging).

lethalman commented on nixos/modules/system/boot/systemd.nix:
```diff
@@ -934,6 +957,12 @@ in
         [Sleep]
       '';
 
+    environment.etc."systemd/coredump.conf".text =
```
Is it possible to make this conditional as well in the mkMerge below?

auntieNeo commented:
So, I tried to rebase this work onto master recently, and most of the config file stuff that used to be in `systemd.nix` was moved out. So I moved the coredump stuff completely out of `systemd.nix` and now it is in its own module. This should be much easier to merge anyway, and my last commit reflects that.

I removed the `--enable-coredump` flag, since it would be good to merge this into master before it diverges again. Please merge this into master sooner than later; this is the kind of functionality I use daily for developing/debugging services.

I also remove the comment suggesting that a reboot would be required, since it appears that changes to the `boot.kernel.sysctl` option affect the system immediately.

lethalman commented:
Agree for the separate coredump nix file, I agree less on having it in systemd/boot. But we can change it at any time. Merging thanks.
